From a5bc6e678d81a4fa1b9ea1796356a5033ecd17e9 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Thu, 29 May 2025 11:35:24 +0300
Subject: [PATCH] [JSW-799] jgs: align progress bits as per dma

This patch is to align with dma sw status bit
sequence in next stage fw bringup progress
bits enum.
In Gaudi3 DMA sw status bits only used in boot
desc progress bits enum. Here, we started using
same for next stage fw image bringup also.

Change-Id: Ie72db7b1fd8fa30f938ea5f4b0569ffdce032ab4
---

diff --git a/arch/jgs/arc/progress_bits.h b/arch/jgs/arc/progress_bits.h
index 83ceb27..5db418d 100644
--- a/arch/jgs/arc/progress_bits.h
+++ b/arch/jgs/arc/progress_bits.h
@@ -9,7 +9,9 @@
 enum boot_rom_progress {
 	PROGRESS_INIT,
 	PROGRESS_INIT_ARC1_ICCM_INIT_FAILED,
-	PROGRESS_INIT_SPI_MANIFEST_0_EXCEEDS_MAX_SIZE,
+
+
+	PROGRESS_INIT_SPI_MANIFEST_0_EXCEEDS_MAX_SIZE, /* boot_rom_boot_desc_progress */
 	PROGRESS_INIT_SPI_MANIFEST_0_DMA_START,
 	PROGRESS_INIT_SPI_MANIFEST_0_CRC_FAIL,
 	PROGRESS_INIT_SPI_MANIFEST_0_UNALIGNED_DEST,
@@ -21,12 +23,14 @@
 	PROGRESS_INIT_SPI_MANIFEST_0_HDR_MAGIC_FAIL,
 	PROGRESS_INIT_SPI_MANIFEST_0_INVALID_IMG_OFFSET,
 	PROGRESS_INIT_SPI_MANIFEST_0_VERIFICATION_FAIL,
+
 	PROGRESS_INIT_SPI_MANIFEST_0_INFO_SET_FAIL,
 	PROGRESS_INIT_SPI_MANIFEST_0_DATA_SET_FAIL,
 	PROGRESS_INIT_SPI_MANIFEST_0_INFO_GET_FAIL,
 	PROGRESS_INIT_SPI_MANIFEST_0_DATA_GET_FAIL,
 	PROGRESS_INIT_SPI_MANIFEST_0_FAIL,
-	PROGRESS_INIT_SPI_MANIFEST_1_EXCEEDS_MAX_SIZE,
+
+	PROGRESS_INIT_SPI_MANIFEST_1_EXCEEDS_MAX_SIZE, /* boot_rom_boot_desc_progress */
 	PROGRESS_INIT_SPI_MANIFEST_1_DMA_START,
 	PROGRESS_INIT_SPI_MANIFEST_1_CRC_FAIL,
 	PROGRESS_INIT_SPI_MANIFEST_1_UNALIGNED_DEST,
@@ -38,11 +42,13 @@
 	PROGRESS_INIT_SPI_MANIFEST_1_HDR_MAGIC_FAIL,
 	PROGRESS_INIT_SPI_MANIFEST_1_INVALID_IMG_OFFSET,
 	PROGRESS_INIT_SPI_MANIFEST_1_VERIFICATION_FAIL,
+
 	PROGRESS_INIT_SPI_MANIFEST_1_INFO_SET_FAIL,
 	PROGRESS_INIT_SPI_MANIFEST_1_DATA_SET_FAIL,
 	PROGRESS_INIT_SPI_MANIFEST_1_INFO_GET_FAIL,
 	PROGRESS_INIT_SPI_MANIFEST_1_DATA_GET_FAIL,
 	PROGRESS_INIT_SPI_MANIFEST_1_FAIL,
+
 	PROGRESS_INIT_SPI_VALID_MANIFEST_INDEX_FAIL,
 	PROGRESS_INIT_EFUSE_INIT_FAIL,
 	PROGRESS_INIT_EFUSE_SPI,
@@ -71,32 +77,37 @@
 	PROGRESS_SPI_MANIFEST_1_VERIFY_KEY_HASH_FAIL,
 	PROGRESS_SPI_MANIFEST_1_VERIFY_AUTH_FAIL,
 	PROGRESS_SPI_MANIFEST_1_VERIFY_SUCCESS,
+
+	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG, /* boot_rom_bringup_proggress */
 	PROGRESS_BRINGUP_PCI_PRI,
+	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST,
+	PROGRESS_BRINGUP_PCI_PRI_INVALID_IMG_SIZE,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE,
+	PROGRESS_BRINGUP_PCI_PRI_INVALID_OFFSET, /* TODO: free */
+	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR,
+	PROGRESS_BRINGUP_PCI_PRI_DMA_FINISH,
 	PROGRESS_BRINGUP_PCI_PRI_NO_TMG_DESC,
 	PROGRESS_BRINGUP_PCI_PRI_LOADADDR_INVALID,
 	PROGRESS_BRINGUP_PCI_PRI_STARTADDR_INVALID,
-	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG,
-	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR,
-	PROGRESS_BRINGUP_PCI_PRI_INVALID_IMG_SIZE,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE,
-	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR,
-	PROGRESS_BRINGUP_PCI_PRI_DMA,
 	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL,
 	PROGRESS_BRINGUP_PCI_PRI_VERIFY_SUCCESS,
+
+	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG, /* boot_rom_bringup_proggress */
 	PROGRESS_BRINGUP_PCI_SEC,
+	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST,
+	PROGRESS_BRINGUP_PCI_SEC_INVALID_IMG_SIZE,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE,
+	PROGRESS_BRINGUP_PCI_SEC_INVALID_OFFSET, /* TODO: free */
+	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR,
+	PROGRESS_BRINGUP_PCI_SEC_DMA_FINISH,
 	PROGRESS_BRINGUP_PCI_SEC_NO_IMG_DESC,
 	PROGRESS_BRINGUP_PCI_SEC_LOADADDR_INVALID,
 	PROGRESS_BRINGUP_PCI_SEC_STARTADDR_INVALID,
-	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG,
-	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR,
-	PROGRESS_BRINGUP_PCI_SEC_INVALID_IMG_SIZE,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE,
-	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR,
-	PROGRESS_BRINGUP_PCI_SEC_DMA,
 	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL,
 	PROGRESS_BRINGUP_PCI_SEC_VERIFY_SUCCESS,
+
 	PROGRESS_BOOT_ROM_TPM,
 	PROGRESS_BOOT_ROM_TPM_INIT_FAIL,
 	PROGRESS_BOOT_ROM_TPM_STARTUP_FAIL,
diff --git a/arch/jgs/arc/status.h b/arch/jgs/arc/status.h
index 68a1649..7f3a1b3 100644
--- a/arch/jgs/arc/status.h
+++ b/arch/jgs/arc/status.h
@@ -6,23 +6,25 @@
 #ifndef _STATUS_H_
 #define _STATUS_H_
 
+#include <stdint.h>
 #include <progress_bits.h>
 #include <mem_io.h>
 
 #define BOOT_ROM_STATUS_REG_SZ_BITS	32
 
 enum boot_rom_bringup_proggress {
+	PROGRESS_BRINGUP_TOO_BIG,
 	PROGRESS_BRINGUP_START,
+	PROGRESS_BRINGUP_CRC_FAIL,
+	PROGRESS_BRINGUP_UNALIGNED_DEST,
+	PROGRESS_BRINGUP_INVALID_IMG_SIZE,
+	PROGRESS_BRINGUP_UNALIGNED_IMG_SIZE,
+	PROGRESS_BRINGUP_INVALID_OFFSET,/* TODO: free */
+	PROGRESS_BRINGUP_DMA_ERROR,
+	PROGRESS_BRINGUP_DMA_FINISH,//
 	PROGRESS_BRINGUP_NO_IMG_DESC,
 	PROGRESS_BRINGUP_LOADADDR_INVALID,
 	PROGRESS_BRINGUP_STARTADDR_INVALID,
-	PROGRESS_BRINGUP_TOO_BIG,
-	PROGRESS_BRINGUP_CRC_FAIL,
-	PROGRESS_BRINGUP_UNALIGNED_DEST_ADDR,
-	PROGRESS_BRINGUP_INVALID_IMG_SIZE,
-	PROGRESS_BRINGUP_UNALIGNED_IMG_SIZE,
-	PROGRESS_BRINGUP_DMA_ERROR,
-	PROGRESS_BRINGUP_DMA,
 	PROGRESS_BRINGUP_KEY_AUTH_FAIL,
 	PROGRESS_BRINGUP_VERIFY_SUCCESS,
 };
diff --git a/common/boot_flow.c b/common/boot_flow.c
index 9d6b2cf..26d6a1c 100644
--- a/common/boot_flow.c
+++ b/common/boot_flow.c
@@ -63,7 +63,7 @@
 {
 	struct boot_desc desc = {0};
 	/* TODO: Image status base is complicated. Handle based on img_type */
-	enum boot_rom_progress img_sts_base = PROGRESS_BRINGUP_PCI_PRI;
+	enum boot_rom_progress img_sts_base = PROGRESS_BRINGUP_PCI_PRI_TOO_BIG;
 	enum hl_fw_manifest_type curr_manifest_idx;
 	uint32_t ret = BOOT_ROM_FAIL;
 
@@ -81,7 +81,7 @@
 	if (boot_rom_second_image_enabled() &&
 		(!!boot_rom_global.fw_upd_switch_manifest)) {
 		primary_manifest_idx = HL_FW_SPI_MANIFEST_1;
-		img_sts_base = PROGRESS_BRINGUP_PCI_SEC;
+		img_sts_base = PROGRESS_BRINGUP_PCI_SEC_TOO_BIG;
 	}
 
 	curr_manifest_idx = primary_manifest_idx;
@@ -102,7 +102,7 @@
 		curr_manifest_idx = (primary_manifest_idx + 1) % HL_FW_SPI_MANIFEST_MAX_NUM;
 
 		/* TODO: What to do if fw upd switch enabled and flow comes here? */
-		img_sts_base = PROGRESS_BRINGUP_PCI_SEC;
+		img_sts_base = PROGRESS_BRINGUP_PCI_SEC_TOO_BIG;
 
 		ret = boot_flow_prefetch_img_sub(curr_manifest_idx, img_type, &desc, img_sts_base);
 		if (!ret) {
@@ -229,7 +229,7 @@
 
 bool boot_rom_boot_flow_is_ppboot_valid(enum boot_rom_manifests rom_manifest_type)
 {
-	enum boot_rom_progress status_base = PROGRESS_BRINGUP_PCI_PRI;
+	enum boot_rom_progress status_base = PROGRESS_BRINGUP_PCI_PRI_TOO_BIG;
 	enum hl_fw_manifest_type man_idx = primary_manifest_idx;
 
 	if (!boot_rom_boot_flow_is_manifest_valid(rom_manifest_type))
@@ -242,7 +242,7 @@
 		man_idx = (primary_manifest_idx + 1) % HL_FW_SPI_MANIFEST_MAX_NUM;
 
 	if (man_idx == HL_FW_SPI_MANIFEST_1)
-		status_base = PROGRESS_BRINGUP_PCI_SEC;
+		status_base = PROGRESS_BRINGUP_PCI_SEC_TOO_BIG;
 
 	if (boot_rom_verify_fw_image_hash384(man_idx, HL_FW_IMG_PPBOOT)) {
 		boot_rom_status_update_bringup(status_base, PROGRESS_BRINGUP_VERIFY_SUCCESS);
